### PR TITLE
Fix magento/magento-coding-standard#214 with class not found

### PR DESCRIPTION
**Pull request by edspc**
Just copied the original classes from the framework to fix the problem.

### CODE REVIEW

File: Magento2/Helpers/Tokenizer/AbstractTokenizer.php [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    /**
+     * Set string for tokenize
+     *
+     * @param string $value
+     * @return void
+     */
+    public function setString($value)
+    {
+        //phpcs:ignore Magento2.Functions.DiscouragedFunction
```
Comment (edspc) [inline Magento2/Helpers/Tokenizer/AbstractTokenizer.php:97]: Additionally added a line for the pipeline.

**Comment by ihor-sviziev** [comment]
Hi @edspc,
Could you provide some info, how to reproduce the issue? 
So far, the issue isn't reproducing on the develop branch while testing this repo via phpcs using `vendor/bin/phpcs --standard=Magento2 Magento2/ --extensions=php`:
https://github.com/magento/magento-coding-standard/runs/3252930009?check_suite_focus=true
image

**Comment by edspc** [comment]
Hi, @ihor-sviziev,
We have faced that fatal exception while running PHPCS for a .phtml file containing the following code:
```
{{getBlockLabel(type, block)}}
```
Currently, **HtmlDirectiveSniff** in develop branch has two references for non-existent classes ([one](https://github.com/magento/magento-coding-standard/blob/fd643e8c6772f73c7188f8188426e7c30eea6013/Magento2/Sniffs/Html/HtmlDirectiveSniff.php#L188) & [two](https://github.com/magento/magento-coding-standard/blob/fd643e8c6772f73c7188f8188426e7c30eea6013/Magento2/Sniffs/Html/HtmlDirectiveSniff.php#L165)).

**Review by ihor-sviziev — APPROVED** [review]
✔ Approved

Before:
```
$ vendor/bin/phpcs --standard=Magento2 /var/www/html/magento2/dev/tests/static/framework/tests/unit/testsuite/Magento/Sniffs/Html/_files --extensions=html
PHP Fatal error:  Uncaught Error: Class 'Magento2\Sniffs\Html\Template\Tokenizer\Variable' not found in /var/www/html/magento-coding-standard/Magento2/Sniffs/Html/HtmlDirectiveSniff.php:188
Stack trace:
#0 /var/www/html/magento-coding-standard/Magento2/Sniffs/Html/HtmlDirectiveSniff.php(83): Magento2\Sniffs\Html\HtmlDirectiveSniff->validateVariableUsage()
#1 /var/www/html/magento-coding-standard/Magento2/Sniffs/Html/HtmlDirectiveSniff.php(63): Magento2\Sniffs\Html\HtmlDirectiveSniff->processIfDirectives()
#2 /var/www/html/magento-coding-standard/vendor/squizlabs/php_codesniffer/src/Files/File.php(498): Magento2\Sniffs\Html\HtmlDirectiveSniff->process()
#3 /var/www/html/magento-coding-standard/vendor/squizlabs/php_codesniffer/src/Files/LocalFile.php(92): PHP_CodeSniffer\Files\File->process()
#4 /var/www/html/magento-coding-standard/vendor/squizlabs/php_codesniffer/src/Runner.php(630): PHP_CodeSniffer\Files\LocalFile->process()
#5 /var/www/html/m in /var/www/html/magento-coding-standard/Magento2/Sniffs/Html/HtmlDirectiveSniff.php on line 188
```


After:
```
$ vendor/bin/phpcs --standard=Magento2 /var/www/html/magento2/dev/tests/static/framework/tests/unit/testsuite/Magento/Sniffs/Html/_files --extensions=html

FILE: /var/www/html/magento2/dev/tests/static/framework/tests/unit/testsuite/Magento/Sniffs/Html/_files/test-html-directive.html
----------------------------------------------------------------------------------------------------------------------------------------
FOUND 20 ERRORS AFFECTING 1 LINE
----------------------------------------------------------------------------------------------------------------------------------------
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "foo.badif().bad"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "foo.badif3()"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "foo.badDepend().bad"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "foo.badForLoop()"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "foo.bad()"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "foo.bad()|alsobad"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "foo.bad($bad.param())"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "foo.bad.baz()"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "foo.bad.trans()"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "bad.bad()"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "bad.multiline()"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "foo.badif2()"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "foo.badif4()"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "foo.badif5()"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "foo.baddepend2()"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
   |       | Found "foo.badFor()"
 1 | ERROR | Template directives may not invoke methods. Only scalar array access is allowed.
 
 ...
 
 FILE: /var/www/html/magento2/dev/tests/static/framework/tests/unit/testsuite/Magento/Sniffs/Html/_files/test-html-binding.html
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 6 ERRORS AFFECTING 1 LINE
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 1 | ERROR | Variables/functions used for HTML binding must have UnsanitizedHtml suffix - "testError()" doesn't,
   |       | consider using text binding if the value is supposed to be text
 1 | ERROR | Variables/functions used for HTML binding must have UnsanitizedHtml suffix - "test.getSomething().value.error()" doesn't,
   |       | consider using text binding if the value is supposed to be text
 1 | ERROR | Variables/functions used for HTML binding must have UnsanitizedHtml suffix - "bind_stuff(1, 2)" doesn't,
   |       | consider using text binding if the value is supposed to be text
 1 | ERROR | Variables/functions used for HTML binding must have UnsanitizedHtml suffix - "testError()" doesn't,
   |       | consider using text binding if the value is supposed to be text
 1 | ERROR | Variables/functions used for HTML binding must have UnsanitizedHtml suffix - "test.getSomething().value.error(1)" doesn't,
   |       | consider using text binding if the value is supposed to be text
 1 | ERROR | Variables/functions used for HTML binding must have UnsanitizedHtml suffix - "bind_stuff()" doesn't,
   |       | consider using text binding if the value is supposed to be text
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Time: 42ms; Memory: 8MB
```

**Comment by sivaschenko** [comment]
@magento import pr to magento-commerce/magento-coding-standard

**Comment by magento-engcom-team** [comment]
@sivaschenko the pull request successfully imported.

**Pull request merged**